dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -2256,11 +2341,51 @@ private void buildCase(CaseTree tree, int index) {
           scan(stmt, null);
         }
       } else {
-        scan(TreeUtils.caseTreeGetBody(tree), null);
+        Tree bodyTree = TreeUtils.caseTreeGetBody(tree);
+        if (bodyTree instanceof ExpressionTree) {
```

Comment:
This might be a third different test for the "switch statement vs switch expression" concept.  I think consistency would make the code easier to read.